pxtpy/pydecompiler.ts

```diff
@@ -769,13 +715,21 @@ namespace pxt.py {
                 stmts.concat(exp)
             }
             if (stmts.length) {
+                // global or nonlocal declerations
+                let globals = scopeLookup.getExplicitGlobals(s)
+                    .map(g => getName(g))
+                if (globals && globals.length)
+                    stmts.unshift(`global ${globals.join(", ")}`)
+                let nonlocals = scopeLookup.getExplicitNonlocals(s)
+                    .map(n => getName(n))
+                if (nonlocals && nonlocals.length)
+                    stmts.unshift(`nonlocal ${nonlocals.join(", ")}`)
```

Comment:
Note that even though this PR is somewhat large, it all comes down to these few lines. These are the only lines that actually change the way python code is generated.